toolchain/lower/file_context.cpp

```diff
@@ -390,6 +423,24 @@ auto FileContext::BuildFunctionBody(SemIR::FunctionId function_id,
                                     SemIR::SpecificId specific_id) -> void {
   const auto& body_block_ids = function.body_block_ids;
   CARBON_DCHECK(llvm_function, "LLVM Function not found when lowering body.");
+
+  if (function.cpp_decl) {
+    clang::FunctionDecl* cpp_def = function.cpp_decl->getDefinition();
+    CARBON_DCHECK(cpp_def, "No Clang function body found during lowering");
+
+    // Create the LLVM function (`CodeGenModule::GetOrCreateLLVMFunction()`) so
+    // that code generation (`CodeGenModule::EmitGlobal()`) would see this
+    // function name (`CodeGenModule::getMangledName()`), and will generate its
+    // definition.
+    CARBON_DCHECK(
+        cpp_code_generator_->GetAddrOfGlobal(clang::GlobalDecl(cpp_def),
+                                             /*isForDefinition=*/false));
```

Comment:
Please don't put code with side-effects inside a CHECK or especially a DCHECK. Instead, call the function outside the CHECK and separately test its result.

Reply:
Makes sense, thanks, done.